app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt

```diff
@@ -47,83 +45,69 @@ class UserTokensPolling(
     private val solTokensFlow = MutableStateFlow<List<Token.Active>?>(null)
     private val ethTokensFlow = MutableStateFlow<List<Token.Eth>?>(null)
 
-    private val isTokensRateFetched = AtomicBoolean(false)
     private var refreshJob: Job? = null
 
     fun shareTokenPollFlowIn(scope: CoroutineScope): StateFlow<Pair<List<Token.Active>?, List<Token.Eth>?>> =
         solTokensFlow.combine(ethTokensFlow) { sol, eth ->
             sol to eth
-        }.stateIn(scope, SharingStarted.WhileSubscribed(), Pair(emptyList(), emptyList()))
+        }.stateIn(scope, SharingStarted.WhileSubscribed(), Pair(null, null))
 
     suspend fun refresh() {
-        isTokensRateFetched.set(false)
         solTokensFlow.emit(null)
         ethTokensFlow.emit(null)
-        startPolling()
+        initTokens()
     }
 
-    fun startPolling() {
+    fun initTokens() {
+        launch {
+            try {
+                val tokensBefore = userInteractor.getUserTokens()
+                val solTokens = async { fetchSolTokens() }
+                val ethTokens = async { fetchEthereumTokens() }
+                userInteractor.loadUserRates(solTokens.await())
```

Comment:
no, we should show user tokens asap without waiting for token rates

Reply:
okay, I will fix It now